docs/Antora/modules/howto_guides/pages/howto_configure_instance_providers.adoc

```diff
@@ -0,0 +1,206 @@
+= Instance Configuration with Providers
+
+A Config Provider is an interface that enriches the input options dictionary with values from external sources such as a JSON file, environment variables, or command line arguments. A Config Provider can be assigned to the instance builder through the method `addConfigProvider`. This method does not override the options dictionary; instead, it retains the provider in the builder to augment options while building the instance.
+
+**Related articles:**
+
+- xref:howto_configure_instance.adoc#instance[Architecture - Instance]
+
+== Instance Configuration with JSON Provider
+
+An instance can be configured from a JSON file. An example of a JSON file is listed below. JSON keys are not case-sensitive, so the keys "MODULEMANAGER", "modulemanager", and "MoDuLeMaNaGeR" are recognized as the same key. If a value has the wrong type, or there are additional key-value pairs, they will be ignored. If a provider is overriding a list value, the old list items will be erased.
+[tabs]
+====
+JSON::
++
+[source,json]
+----
+{
+  "ModuleManager": {
+    "ModulesPath": ""
+  },
+  "Scheduler": {
+    "WorkersNum": 0
+  },
+  "Logging": {
+    "GlobalLogLevel": 7
+  },
+  "RootDevice": {
+    "DefaultLocalId": "",
+    "Connection": ""
+  },
+  "Modules": {
+    "SomeModule1": {
+        "option1": "value"
+    },
+    "SomeModule2": {
+        "enableOption2": true,
+        "listOfSomething": [
+            1,
+            2,
+            3,
+            4
+        ]
+    }
+  }
+}
+----
+====
+To use configuration from a JSON file, a developer can create a `JsonConfigProvider` with the file path. If the file path is not configured, the provider will attempt to read the `OPENDAQ_CONFIG_PATH` environment variable. If this variable is also not set, the provider will try to read `opendaq-config.json` from the same folder from which the file was run.
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+
+int main(int argc, const char* argv[])
+{
+    const StringPtr configPath = "opendaq-config.json";
+    const ConfigProviderPtr configProvider = JsonConfigProvider(configPath);
+    const InstanceBuilderPtr instanceBuilder = InstanceBuilder().addConfigProvider(configProvider);
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder();
+
+    return 0;
+}
```

Comment:
Code snippets such as these should be added to the documentation tests for validation and testing.

Both for c++ and python